parquet-column/src/main/java/org/apache/parquet/column/statistics/Statistics.java

```diff
@@ -73,6 +73,70 @@ public Builder withNumNulls(long numNulls) {
     }
   }
 
+  // Builder for FLOAT type to handle special cases of min/max values like NaN, -0.0, and 0.0
+  private static class FloatBuilder extends Builder {
+    public FloatBuilder(PrimitiveType type) {
+      super(type);
+      assert type.getPrimitiveTypeName() == PrimitiveTypeName.FLOAT;
+    }
+
+    @Override
+    public Statistics<?> build() {
+      FloatStatistics stats = (FloatStatistics) super.build();
+      if (stats.hasNonNullValue()) {
+        Float min = stats.genericGetMin();
+        Float max = stats.genericGetMax();
+        // Drop min/max values in case of NaN as the sorting order of values is undefined for this case
+        if (min.isNaN() || max.isNaN()) {
+          stats.setMinMax(0.0f, 0.0f);
+          ((Statistics<?>) stats).hasNonNullValue = false;
+        } else {
+          // Updating min to -0.0 and max to +0.0 to ensure that no 0.0 values would be skipped
+          if (min == 0.0f) {
+            stats.setMinMax(-0.0f, max);
+            min = -0.0f;
```

Comment:
(nit) In my opinion,
```
min = -0.0f;
stats.setMinMax(min, max);
```
would be cleaner. Similar for the max case (although max is not used any more in this function).